ipywebrtc/webrtc.py

```diff
@@ -225,6 +233,61 @@ def save(self, filename=None):
     _video_src = Unicode('').tag(sync=True)
 
 
+
+# monkey patch, same as https://github.com/jupyter-widgets/ipywidgets/pull/2146
+if 'from_json' not in widgets.Image.value.metadata:
+    widgets.Image.value.metadata['from_json'] = lambda js, obj: None if js is None else js.tobytes()
+
+@register
+class MediaImageRecorder(DOMWidget):
+    """Creates a recorder which allows to grab an Image from a MediaStream widget.
+    """
+    _model_module = Unicode('jupyter-webrtc').tag(sync=True)
+    _view_module = Unicode('jupyter-webrtc').tag(sync=True)
+    _model_name = Unicode('MediaImageRecorderModel').tag(sync=True)
+    _view_name = Unicode('MediaImageRecorderView').tag(sync=True)
+    _view_module_version = Unicode(semver_range_frontend).tag(sync=True)
+    _model_module_version = Unicode(semver_range_frontend).tag(sync=True)
+
+    stream = Instance(MediaStream, allow_none=True).tag(sync=True, **widget_serialization)
+    image = Instance(Image, allow_none=True).tag(sync=True, **widget_serialization)
+    filename = Unicode('stream-image').tag(sync=True)
```

Comment:
Yes, ImageRecorder and VideoRecorder would be simpler. Also, this is asking for a Video widget, similar as the ipywidgets.Image widget..

Reply:
To have consistent naming, what about:
 * VideoStream (not that we're not using VideoMediaStream, so we're skipping Media)
 * StreamVideoRecorder
 * StreamImageRecorder
 * StreamAudioRecorder

Reply:
I'm also ok calling it: MediaStream<name>Recorder

Reply:
I think `MediaStreamRecorder` is a bit better than `VideoStream`, so that it's clear that you can record any `MediaStream` with it, not only `VideoStream`s

Reply:
Sorry I got confused, a bit tired :P 
I think MediaStreamVideoRecorder and MediaStreamImageRecorder are too long though it's clearer that it records a MediaStream. 
What did you want to call MediaStreamRecorder exactly?

Those three were good:
- StreamVideoRecorder
- StreamImageRecorder
- StreamAudioRecorder
or maybe:
- VideoStreamRecorder
- ImageStreamRecorder
- AudioStreamRecorder

But it's unclear that it records a MediaStream

Reply:
Yes, the names aren't clear. The thing is that MediaRecorder is really a thing, a javascript object, and if we want to mimic the JS API, it makes sense to have such a widget as well, hence its name now.

Reply:
To stay close to the JS API, I've decided to keep the names as it is.